st_package_reviewer/check/file/__init__.py

```diff
@@ -1,13 +1,14 @@
 import functools
 import itertools
 from pathlib import Path
+import ast
 
 from .. import Checker, find_all
 
 __all__ = ('FileChecker', 'get_checkers')
 
 
-class FileChecker(Checker):
+class FileChecker(Checker, ast.NodeVisitor):
```

Comment:
I actually agree with having an `ASTChecker` derived from `FileChecker` instead; I'll put the AST checks in a subfolder